packages/edit-site/src/components/page-patterns/duplicate-menu-item.js

```diff
@@ -2,217 +2,141 @@
 	const history = useHistory();
-	const existingTemplateParts = useExistingTemplateParts();
-	const { patternCategories } = usePatternCategories();
 
-	async function createTemplatePart() {
-		try {
-			const copiedTitle = sprintf(
-				/* translators: %s: Existing template part title */
-				__( '%s (Copy)' ),
-				item.title
-			);
-			const title = getUniqueTemplatePartTitle(
-				copiedTitle,
-				existingTemplateParts
-			);
-			const slug = getCleanTemplatePartSlug( title );
-			const { area, content } = item.templatePart;
-
-			const result = await saveEntityRecord(
-				'postType',
-				TEMPLATE_PART_POST_TYPE,
-				{ slug, title, content, area },
-				{ throwOnError: true }
-			);
-
-			createSuccessNotice(
-				sprintf(
-					// translators: %s: The new template part's title e.g. 'Call to action (copy)'.
-					__( '"%s" duplicated.' ),
-					item.title
-				),
-				{
-					type: 'snackbar',
-					id: 'edit-site-patterns-success',
-				}
-			);
-
-			history.push( {
-				postType: TEMPLATE_PART_POST_TYPE,
-				postId: result?.id,
-				categoryType: TEMPLATE_PART_POST_TYPE,
-				categoryId,
-			} );
+	const isTemplatePart = item.type === TEMPLATE_PART_POST_TYPE;
 
-			onClose();
-		} catch ( error ) {
-			const errorMessage =
-				error.message && error.code !== 'unknown_error'
-					? error.message
-					: __(
-							'An error occurred while creating the template part.'
-					  );
-
-			createErrorNotice( errorMessage, {
+	async function onTemplatePartSuccess( templatePart ) {
+		createSuccessNotice(
+			sprintf(
+				// translators: %s: The new template part's title e.g. 'Call to action (copy)'.
+				__( '"%s" duplicated.' ),
+				item.title
+			),
+			{
 				type: 'snackbar',
-				id: 'edit-site-patterns-error',
-			} );
-			onClose();
-		}
-	}
-
-	async function findOrCreateTerm( term ) {
-		try {
-			const newTerm = await saveEntityRecord(
-				'taxonomy',
-				'wp_pattern_category',
-				{
-					name: term.label,
-					slug: term.name,
-					description: term.description,
-				},
-				{
-					throwOnError: true,
-				}
-			);
-			invalidateResolution( 'getUserPatternCategories' );
-			return newTerm.id;
-		} catch ( error ) {
-			if ( error.code !== 'term_exists' ) {
-				throw error;
+				id: 'edit-site-patterns-success',
 			}
+		);
 
-			return error.data.term_id;
-		}
+		history.push( {
+			postType: TEMPLATE_PART_POST_TYPE,
+			postId: templatePart?.id,
+			categoryType: TEMPLATE_PART_POST_TYPE,
+			categoryId,
+		} );
+
+		onClose();
 	}
 
-	async function getCategories( categories ) {
-		const terms = categories.map( ( category ) => {
-			const fullCategory = patternCategories.find(
-				( cat ) => cat.name === category
-			);
-			if ( fullCategory.id ) {
-				return fullCategory.id;
+	function onPatternSuccess( { pattern } ) {
+		createSuccessNotice(
+			sprintf(
+				// translators: %s: The new pattern's title e.g. 'Call to action (copy)'.
+				__( '"%s" duplicated.' ),
+				pattern.title
```

Comment:
This shows the infamous `[object object]` in the snackbar as shown in the video too.

Reply:
Thanks, I completely missed that but just ran into it while adding the duplicate pattern command. I'll fix it up tomorrow.